lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -33,7 +35,27 @@ struct FirMemory {
   size_t readLatency;
   size_t writeLatency;
   size_t readUnderWrite;
+  size_t writeUnderWrite;
+  llvm::EquivalenceClasses<unsigned> writeSets;
 };
+
+class SymbolDigger {
+  DenseMap<Attribute, SmallVector<Operation *, 1>> internal;
+
+public:
+  SymbolDigger(ModuleOp module) {
+    module.walk([&](Operation *op) {
```

Comment:
instances can only exist at the top level, so no need to use "walk" which iterates into if's and stuff like that.

I think you can use something like `for (auto instance : module.getBodyBlock().getOps<hw::InstanceOp>())`

Reply:
actually, that wouldn't get instances in ifdef's.  I'm not sure if you care.

Reply:
Another question: should this use `SymbolCache` in `Dialect/HW/HWOps.h`?

Reply:
Oh wait.- ModuleOp in this case is the top level module, not hw::Module.  We have utilities for this, we should use one of the existing things that does this digging, potentially move this to a utility method or constructor on SymbolCache.

Reply:
This `SymbolDigger` is now gone. Clock information is found during `LowerToHW` when the `firrtl.mem` is being examined.  There is then no need for any symbol chasing.